File: selection-sort/selection-sort.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <time.h>

#define length 20

int i, x;

void embaralha_baralho(int *v)
{
    srand(time(NULL)); // Seed para geração de números aleatórios
    for (int i = 0; i < length; i++)
    {
        v[i] = rand() % 1000; // Preenche o baralho com valores aleatórios de 0 a 999
    }
}

void desembaralha_baralho(int *v)
{
    int troca = true;
    do
    {
        troca = false;
        for (i = 0; i < length; i++)
        {
            for (x = i + 1; x < length; x++)
            {
                if (v[i] > v[x])
                {
                    int a = v[x];
                    v[x] = v[i];
                    v[i] = a;
                    troca = true;
                }
            }
        }
    } while (troca);
}

void posicao_baralho(int *v)
{
    for (int i = 0; i < length; i++)
    {
        printf("Posicao %d: %d\n", i, v[i]);
    }
}

int main()
{
    int vet[length];

    embaralha_baralho(vet);
    desembaralha_baralho(vet);
    posicao_baralho(vet);

    return 0;
}

File: selection-sort/selection-sort.py
from random import randint

def embaralha(v):
    for a in range(len(v)):
        v[a] = randint(0, 100)


def selectionSort(v):
    troca = True
    while troca is True:
        troca = False
        for a in range(0, len(v) - 1, 1):
            for i in range(a + 1, len(v), 1):
                if v[a] > v[i]:
                    aux = v[a]
                    v[a] = v[i]
                    v[i] = aux
                    troca = True


def printa(v):
    for a in range(len(v)):
        print(f"Posicao {a}: {v[a]}")

length = 20

vet = [0] * length

embaralha(vet)

selectionSort(vet)

printa(vet)
